### DISCUSSION TITLE
Can't get to make SMTP work

### DISCUSSION THREAD

**Discussion by Turab** [chatwoot/chatwoot#7017] [Self hosted]
I have setup Chatwoot in Caprover. I setup e-mail channel and it receives converstations but I can't make SMTP work. I don't see any errors in the logs, everything seems pretty ok but the responses made from Chatwoot is never sent from SMTP to the customer. I don't find the docs helpful about this. I have been searching for hours now.

How can I make it work or how can I even diagnose to understand where lies the problem?

**Comment by joseph0063** [comment]
hello, did you follow this
https://www.chatwoot.com/docs/product/channels/email/configure-smtp/

**Comment by Turab** [comment]
Yes I did; but it never shows a sent indicator as shown in the docs:
![image](https://github.com/chatwoot/chatwoot/assets/1780719/f7c7539f-235d-485f-85b5-9482f5d5ff14)

And I never receive the response in the inbox. My SMTP settings are correct, I am already using it in another client. IMAP also uses the same credentials and server and it works in Chatwoot; but SMTP doesn't. I can't see any errors in logs of neither Chatwoot nor Sidekiq. How can I diagnose the problem?

**Comment by Turab** [comment]
This is really odd. I resolved it by setting the SMTP ENV vars only. Saving SMTP configuration on Chatwoot UI doesn't work. (https://www.chatwoot.com/docs/product/channels/email/configure-smtp/ this method doesn't work.) Setting the same variables as env vars makes it work. This is what I had to add:

MAILER_SENDER_EMAIL=channel's mail address
SMTP_ADDRESS=smtp.server.address
SMTP_USERNAME=username(usually full e-mail and usually the same with MAILER_SENDER_EMAIL)
SMTP_PASSWORD=mail password
SMTP_SSL=true

Ok, it is working now but there is still something wrong here:

1- It should respect the settings in channel configuration. Otherwise why bother putting settings there?
2- It shouldn't use a single SMTP environment for all channels. In the end, you can add different mail channels. So the workaround I used here is very wrong when you are going to use e-mail channels more than one.

EDIT: I was too quick to say it is working. Now things get more weird. I can correspond with the customer, they receive the mails from Chatwoot but "forgot password" link on Chatwoot login page doesn't send any e-mail to any agents. And agents don't receive welcome mails when created from the admin panel. So effectively I can't add agents..

**Reply by zz** [reply]
chatwoot is a un-work project

**Comment by ghost** [comment]
Same issues here :/
can someone from the team help?

**Comment by mredodos** [comment]
@sojan-official we need help with SMTP! multiple behavior

**Reply by zz** [reply]
looks like chatwoot don't support this?

**Comment by zz** [comment]
me too. the index SMTP can't work

**Reply by scmmishra** [reply]
Hey what error are you facing exactly? We will need more information to help you out.

**Reply by Turab** [reply]
> Hey what error are you facing exactly? We will need more information to help you out.

I think the thread is very clear. Chatwoot doesn't send replies to participants via SMTP. You can reply to the tickets, but "sent" indicator never appears on Chatwoot. The participant at the other end doesn't receive any mails for this ticket. SMTP settings are correct. You can see more details in my previous comment:

> This is really odd. I resolved it by setting the SMTP ENV vars only. Saving SMTP configuration on Chatwoot UI doesn't work. (https://www.chatwoot.com/docs/product/channels/email/configure-smtp/ this method doesn't work.) Setting the same variables as env vars makes it work. This is what I had to add:
> 
> MAILER_SENDER_EMAIL=channel's mail address SMTP_ADDRESS=smtp.server.address SMTP_USERNAME=username(usually full e-mail and usually the same with MAILER_SENDER_EMAIL) SMTP_PASSWORD=mail password SMTP_SSL=true
> 
> Ok, it is working now but there is still something wrong here:
> 
> 1- It should respect the settings in channel configuration. Otherwise why bother putting settings there? 2- It shouldn't use a single SMTP environment for all channels. In the end, you can add different mail channels. So the workaround I used here is very wrong when you are going to use e-mail channels more than one.
> 
> EDIT: I was too quick to say it is working. Now things get more weird. I can correspond with the customer, they receive the mails from Chatwoot but "forgot password" link on Chatwoot login page doesn't send any e-mail to any agents. And agents don't receive welcome mails when created from the admin panel. So effectively I can't add agents..

**Comment by mmojadad** [comment]
Hi
In my side SMTP not work as expect , Chat transcripts not send to visitors, But For example new agent welcome email sent, With same SMTP setting!

**Comment by NilsLP25** [comment]
I have the same problem; I've set everything up correctly, but the system isn't sending emails via SMTP.
